interceptors/protovalidate/protovalidate.go

```diff
@@ -14,9 +14,13 @@ import (
 	"google.golang.org/protobuf/proto"
 )
 
+type Validator interface {
+	Validate(proto.Message) error
+}
+
 // UnaryServerInterceptor returns a new unary server interceptor that validates incoming messages.
 // If the request is invalid, clients may access a structured representation of the validation failure as an error detail.
-func UnaryServerInterceptor(validator *protovalidate.Validator, opts ...Option) grpc.UnaryServerInterceptor {
+func UnaryServerInterceptor(validator Validator, opts ...Option) grpc.UnaryServerInterceptor {
```

Comment:
This is actually a breaking change, and I'm not sure we need to make this change? If you want to propose this, please open an issue. If you remove this change I'm happy to merge the rest of this PR.

Reply:
Removed this commit.